app/main.py

```diff
@@ -1,2 +1,51 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table: list = [None] * self.capacity
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        if self.length >= self.capacity * 2 // 3:
+            self.resize()
+
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+        while True:
+            if self.hash_table[index] is None:
+                self.hash_table[index] = [hash_key, key, value]
+                self.length += 1
+                break
+            if (
+                self.hash_table[index][1] == key
+                and self.hash_table[index][0] == hash_key
```

Comment:
```suggestion
                self.hash_table[index][0] == hash_key
                and self.hash_table[index][1] == key
```

Reply:
it's better to check hash values first because comparing of the integers could be much faster than complex keys such as tuples. and if the first check fails the second would be skipped

Reply:
yes, of course